runtime/doc/api.txt

```diff
@@ -55,6 +54,28 @@ error_types		Possible error types returned by API functions
 External programs ("clients") can use the metadata to discover the |rpc-api|.
 
 ==============================================================================
+API contract                                                     *api-contract*
+
+The API is made of functions and events. Basic clients interact with functions
+like those described at |api-global|. Advanced clients may "attach" in order
```

Comment:
I don't think ui clients are more "advanced". Starting to use event streams (UI is the only current one, there will be more in the future), shouldn't be seen as a step up in complexity, that is something the client simply does when it needs to.